openedx_learning/apps/authoring/components/api.py

```diff
@@ -186,11 +187,27 @@ def create_next_component_version(
             component_id=component_pk,
         )
         # First copy the new stuff over...
```

Comment:
FWIW, this was an oversight on my part in the review. Whenever possible, my intent was for these API functions to be atomic, so that we don't get inconsistent half-writes to the database. (Inconsistent half-writes to the file store are harder to guard against, but I feel better about that since they're idempotent inserts storing file data by hash–so "do it again" results in the same thing and hopefully the worst thing we do is write some unreferenced data.)

Reply:
@ormsbee  should we do a follow-up PR to wrap this function in a transaction?

Reply:
I will

Reply:
😄 Ah, that makes sense. Thank you!